test/integration/smoke/test_network.py

```diff
@@ -1271,3 +1271,225 @@ def test_network_rules_acquired_public_ip(self, value):
+            # Cleanup resources used
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="false")
+    def test_deploy_vm_l2network(self):
+        """Creates an l2 network and verifies user is able to deploy a VM in it"""
+
+        # Validate the following:
+        # 1. Deploys a VM
+        # 2. There are no network services available since this is L2 Network
+
+        self.virtual_machine = VirtualMachine.create(
+            self.apiclient,
+            self.services["virtual_machine"],
+            templateid=self.template.id,
+            serviceofferingid=self.service_offering.id,
+            networkids=self.l2_network.id,
+            zoneid=self.zone.id
+        )
+
+        self.cleanup.insert(0, self.virtual_machine)
+
+        list_vm = list_virtual_machines(
+            self.apiclient,
+            id = self.virtual_machine.id
+        )
+        self.assertEqual(
+            isinstance(list_vm, list),
+            True,
+            "Check if virtual machine is present"
+        )
+
+        self.assertEqual(
+            list_vm[0].nic[0].type,
+            'L2',
+            "Check Correct Network type is available"
+        )
+
+        self.assertFalse(
+            'gateway' in str(list_vm[0].nic[0])
+        )
+
+        self.assertFalse(
+            'ipaddress' in str(list_vm[0].nic[0])
+        )
+
+        return
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="false")
+    def test_delete_network_while_vm_on_it(self):
+        """It verifies the user is not able to delete network which has running vms"""
+
+        # Validate the following:
+        # 1. Deploys a VM
+        # 2. Tries to delete network and expects exception to appear
+
+        self.virtual_machine = VirtualMachine.create(
+            self.apiclient,
+            self.services["virtual_machine"],
+            templateid=self.template.id,
+            serviceofferingid=self.service_offering.id,
+            networkids=self.l2_network.id,
+            zoneid=self.zone.id
+        )
+
+        self.cleanup.insert(0, self.virtual_machine)
+
+        list_vm = list_virtual_machines(
+            self.apiclient,
+            id = self.virtual_machine.id
+        )
+        self.assertEqual(
+            isinstance(list_vm, list),
+            True,
+            "Check if virtual machine is present"
+        )
+
+        try:
+            self.l2_network.delete(self.apiclient)
+        except Exception:
+            pass
+        else:
+            self.fail("Expected an exception to be thrown, failing")
+
+        return
+
+    @attr(tags=["advanced", "advancedns", "smoke"], required_hardware="false")
+    def test_l2network_restart(self):
+        """It verifies the user is not able to delete network which has running vms"""
```

Comment:
Duplicated test description, can we refactor this?

Reply:
Fixed